features/events/daemon/src/main/java/org/opennms/netmgt/eventd/XmlEventProcessor.java

```diff
@@ -0,0 +1,182 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.eventd;
+
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.opennms.core.xml.JaxbUtils;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.EventReceipt;
+import org.opennms.netmgt.xml.event.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+
+/**
+ * Provides the logic and context of execution necessary to actually process a
+ * client's event XML document. When a new stream handler is created and
+ * assigned to an execution context it will unmarshal the remote document. The
+ * events from the remote document are then passed to the registered event
+ * handlers. All successfully processed events are acknowledged to the client by
+ * the generation of an XML event receipt.
+ * 
+ * @author <a href="mailto:weave@oculan.com">Brian Weaver </a>
+ * @author <a href="http;//www.opennms.org">OpenNMS </a>
+ * 
+ */
+public class XmlEventProcessor implements Processor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(XmlEventProcessor.class);
+
+    private EventForwarder m_eventForwarder;
+
+    public EventForwarder getEventForwarder() {
+        return m_eventForwarder;
+    }
+
+    public void setEventForwarder(EventForwarder eventForwarder) {
+        m_eventForwarder = eventForwarder;
+    }
+
+    @Override
+    public void process(Exchange exchange) {
+
+        final InputStream stream = exchange.getIn().getBody(InputStream.class);
+
+        // Unmarshal the XML document
+        Log eventLog = null;
+        try {
+            eventLog = JaxbUtils.unmarshal(Log.class, new InputSource(stream));
+            LOG.debug("Event record converted");
+        } catch (final Throwable e) {
+            LOG.error("Could not unmarshal the XML record", e);
+            // TODO: Throw an exception or something?
```

Comment:
Shouldn't we re-throw the exception here? We can the use Camel to keep track of the count of failed exchanges.